COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md

```diff
@@ -548,6 +548,254 @@ The result is execution of COBOL program CBLDB21 to read the Db2 table and write
+
+```
+//GO.DDName  DD  DSN=data-set-name
+```
+
+### Writing JCL to compile programs
+
+Chances are you will not need to manually write any JCL to compile a program. However, if the catalogued procedure does not provide you with the flexibility you need, you can write your job control statements. Let us take a look and study the following example:
+
+```
+//jobname JOB acctno,name,MSGCLASS=1              (1)
+//stepname EXEC PGM=IGYCRCTL,PARM=(options)       (2)
+//STEPLIB  DD DSNAME=IGY.V6R3M0.SIGYCOMP,DISP=SHR (3)
+//         DD DSNAME=SYS1.SCEERUN,DISP=SHR
+//         DD DSNAME=SYS1.SCEERUN2,DISP=SHR
+//SYSUT1   DD UNIT=SYSALLDA,SPACE=(subparms)      (4)
+//SYSUT2   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT3   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT4   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT5   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT6   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT7   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT8   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT9   DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT10  DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT11  DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT12  DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT13  DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT14  DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSUT15  DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSMDECK DD UNIT=SYSALLDA,SPACE=(subparms)
+//SYSPRINT DD SYSOUT=A                            (5)
+//SYSLIN   DD DSNAME=MYPROG,UNIT=SYSALLDA,        (6)
+//            DISP=(MOD,PASS),SPACE=(subparms)
+//SYSIN    DD DSNAME=dsname,UNIT=device,          (7)
+              VOLUME=(subparms),DISP=SHR
+```
+
+**(1):** The JOB statement indicates the beginning of a job.
+
+**(2):** The EXEC statement specifies that the Enterprise COBOL compiler (IGYCRCTL) is to be invoked.
+
+**(3):** The DD statement here define where the Enterprise COBOL compiler resides. You will need to check with your system programmer regarding where the compiler is installed on your system. Alongside them, the Language Environment SCEERUN and SCEERUN2 data sets must be included in the concatenation unless they are available in the LNKLST.
+
+**(4):** The SYSUT DD statements define the utility data sets that the compiler will use to process the source program. All SYSUT files must be on direct-access storage devices.
+
+**(5):** The SYSPRINT DD statement defines the data set that receives output from compiler options.
+
+**(6):** The SYSLIN DD statement defines the data set that receives output from the OBJECT compiler option.
+
+**(7):** The SYSIN DD statement defines the data set to be used as input to the job step, or in other words, the source code.
```

Comment:
Overall, I like how you organized this section - very clear 👍

Reply:
To be honest, this came straight from the Programming Guide. I am a bit afraid it might be too complex though... but then again, this would be a nice-to-have knowledge.